R/utils_render_latex.R

```diff
@@ -70,6 +70,76 @@ create_table_start_l <- function(data) {
   )
 }
 
+#' Create the heading component of a table
+#'
+#' The table heading component contains the title and possibly a subtitle; if
+#' there are no heading components defined this function will return an empty
+#' string.
+#'
+#' @noRd
+create_heading_component_l <- function(data) {
+
+  # If there is no title or heading component, then return an empty string
+  if (!dt_heading_has_title(data = data)) {
+    return("")
+  }
+
+  heading <- dt_heading_get(data = data)
+  footnotes_tbl <- dt_footnotes_get(data = data)
+  subtitle_defined <- dt_heading_has_subtitle(data = data)
+
+  # Get the footnote marks for the title
+  if ("title" %in% footnotes_tbl$locname) {
+
+    footnote_title_marks <-
+      coalesce_marks(
+        fn_tbl = footnotes_tbl,
+        locname = "title"
+      )
+
+    footnote_title_marks <-
+      footnote_mark_to_latex(mark = footnote_title_marks$fs_id_c)
+
+  } else {
+    footnote_title_marks <- ""
+  }
+
+  # Get the footnote marks for the subtitle
+  if (subtitle_defined && "subtitle" %in% footnotes_tbl$locname) {
+
+    footnote_subtitle_marks <-
+      coalesce_marks(
+        fn_tbl = footnotes_tbl,
+        locname = "subtitle"
+      )
+
+    footnote_subtitle_marks <-
+      footnote_mark_to_latex(mark = footnote_subtitle_marks$fs_id_c)
+
+  } else {
+    footnote_subtitle_marks <- ""
+  }
+
+  title_row <-
```

Comment:
For both `\large` and `\small`, use outer curly braces (maybe a helper function to create these). And remove the newline from the ends of title and subtitle; instead, use a newline at the beginning of subtitle (if there is a subtitle).